### DISCUSSION TITLE
is there any possible to support third party gpt api

### DISCUSSION THREAD

**Discussion by wingeva1986** [danny-avila/LibreChat#190] [Feature Requests & Suggestions]
is there any possible to support third party gpt api, such as poe.com.
and is it nessesary to make a fake openai api to integrity?

**Comment by danny-avila** [comment]
> is there any possible to support third party gpt api, such as poe.com. 

Yes it is, provided there is a library that reverse engineers access. Fortunately there is! This is my big to-do after messing with plugins a bit more. I will focus on Bard first, and then see about poe.

>and is it nessesary to make a fake openai api to integrity?

Can you expand on this point, I'm not sure what you mean

**Comment by wingeva1986** [comment]
The fake OpenAI API handles messages in the same format as the official API does, but is based on a third-party (POE) service.Or to put it another way, just like reverse proxy.

**Comment by danny-avila** [comment]
> The fake OpenAI API handles messages in the same format as the official API does, but is based on a third-party (POE) service.Or to put it another way, just like reverse proxy.

it's not necessary for this project. It is helpful, but that would be pretty involved and beyond the scope of this project. There are some useful repos that do this for LLMs and I will look into integrating them.

For POE or any LLM available through API, I can utilize any service and make a client class that will work.
